gateway/gateway_test.go

```diff
@@ -832,6 +832,17 @@ func TestControlListener(t *testing.T) {
 	ts.RunExt(t, tests...)
 	DoReload()
 	ts.RunExt(t, tests...)
+
+	// Moved here because DoReload overrides it
+	BuildAndLoadAPI(func(spec *APISpec) {
+		spec.Proxy.ListenPath = "/user-api"
+		spec.UseKeylessAccess = true
+	})
+
+	_, _ = ts.Run(t, []test.TestCase{
+		{Path: "/user-api", Code: http.StatusOK},
+		{Path: "/user-api", ControlRequest: true, Code: http.StatusNotFound},
```

Comment:
please add this inside the `tests` variable and remove this block entirely. We are only extending the tests so this block isn't needed.

Reply:
Don't you see my comment there?

Reply:
`// Moved here because DoReload overrides it`

Reply:
I think @gernest meant smth like this:
You more tests cases inside tests object and your test will look as simple as :

``` 
	BuildAndLoadAPI(func(spec *APISpec) {
		spec.Proxy.ListenPath = "/user-api"
		spec.UseKeylessAccess = true
	})

        ts.RunExt(t, tests...)
        doReload()

	// DoReload overrides APIs, needs loading again
	BuildAndLoadAPI(func(spec *APISpec) {
		spec.Proxy.ListenPath = "/user-api"
		spec.UseKeylessAccess = true
	})
        ts.RunExt(t, tests...)
```

Reply:
So basically we verify that API actually loaded before the reload, and still working after reload.

Reply:
@furkansenharputlu ☝️ see @buger  comment  above that is what I meant.

Reply:
Also, a better way will be extending `LoadAPI` to actually allow keep API between reloads.

Reply:
For sure I firstly added inside the tests object. However, after seeing that `DoReload` overrides it, I moved it to there. I am not sure this way is simpler and covers any new thing but I will update as you said.

Reply:
Nope, your way will also not work because `RunExt` fails with user api.

```
func TestDummy(t *testing.T) {
	ts := StartTest(TestConfig{
		sepatateControlAPI: true,
	})
	defer ts.Close()

	BuildAndLoadAPI(func(spec *APISpec) {
		spec.Proxy.ListenPath = "/user-api"
		spec.UseKeylessAccess = true
	})

	tests := []test.TestCase{
		{Path: "/user-api", Code: http.StatusOK},
	}

	ts.RunExt(t, tests...)
}```

This doesn't work.

Reply:
Hmm, it should. Needs investigation.

Reply:
So, I am in favor of merging the current version.